e2e/wdio/core/tests/standard-list.e2e-spec.ts

```diff
@@ -0,0 +1,422 @@
+            checkElementStates(filterAndSortList + listItems, 'focus', 'filter and sort list item');
+            checkElementStates(filterAndSortList + listItems, 'active', 'filter and sort list item');
+        });
+
+        it('should check ability to search', () => {
+            scrollIntoView(filterAndSortList);
+            click(filterAndSortList + searchBar);
+            sendKeys('apple');
+            const resultCount = getElementArrayLength(filterAndSortList + listItems);
+            for (let i = 0; i < resultCount; i++) {
+                expect(getText(filterAndSortList + listItems, i).toLowerCase()).toContain('apple');
+            }
+        });
+
+        it('should check the ability to clear search', () => {
+            const startItemCount = getElementArrayLength(filterAndSortList + listItems);
+
+            scrollIntoView(filterAndSortList);
+            click(filterAndSortList + searchBar);
+            sendKeys('apple');
+
+            expect(getElementArrayLength(filterAndSortList + listItems)).not.toEqual(startItemCount);
+
+            click(filterAndSortList + button);
+
+            expect(getElementArrayLength(filterAndSortList + listItems)).toEqual(startItemCount);
+        });
+
+        it('should check the ability to sort', () => {
+            const itemCount = getElementArrayLength(filterAndSortList + listItems);
+            const originalFirstItem = getText(filterAndSortList + listItems);
+            const originalLastItem = getText(filterAndSortList + listItems, itemCount - 1);
+
+            click(filterAndSortList + button, 2);
+
+            expect(getText(filterAndSortList + listItems)).toBe(originalLastItem);
+            expect(getText(filterAndSortList + listItems, itemCount - 1)).toBe(originalFirstItem);
+
+            click(filterAndSortList + button, 1);
+
+            expect(getText(filterAndSortList + listItems)).toBe(originalFirstItem);
+            expect(getText(filterAndSortList + listItems, itemCount - 1)).toBe(originalLastItem);
+        });
+
+        it('should check the ability to delete an item', () => {
+            const startItemCount = getElementArrayLength(filterAndSortList + listItems);
+
+            click(filterAndSortList + button, 4);
+
+            expect(getElementArrayLength(filterAndSortList + listItems)).toEqual(startItemCount - 1);
+        });
+    });
+
+    describe('secondary item on list example ', function() {
+        it('check list item states', () => {
+            checkElementStates(secondaryItemList + listItems, 'hover', 'secondary item list item');
+            checkElementStates(secondaryItemList + listItems, 'focus', 'secondary item list item');
+            checkElementStates(secondaryItemList + listItems, 'active', 'secondary item list item');
+        });
+    });
+
+    describe('list with icons example', function() {
+        it('check list item states', () => {
```

Comment:
Shouldn't we move scenarios related to visual regression in a separate describe 'Visual regression'?

Reply:
This spec is 95% vusiual regression. With that, I decided to go example by example to make everything easier for everyone. Easier to debug, easier to find which spec to run in case of failure, etc. And having 2 describes for some example blocks would be confusing. So I think the current way is better.

Reply:
Ok, np